tests/test_pyrflx.py

```diff
@@ -592,16 +630,502 @@ def test_ethernet_parse_binary(self) -> None:
+
+        payload = (
+            b"\x45\x00\x00\x2e\x00\x01\x00\x00\x40\x11\x7c\xbc"
+            b"\x7f\x00\x00\x01\x7f\x00\x00\x01\x00\x35\x00\x35"
+            b"\x00\x1a\x01\x4e\x00\x00\x00\x00\x00\x00\x00\x00"
+            b"\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00"
+        )
+
+        self.frame.set("Destination", int("FFFFFFFFFFFF", 16))
+        self.frame.set("Source", int("0", 16))
+        self.frame.set("Type_Length_TPID", int("0800", 16))
+        self.frame.set("Type_Length", int("0800", 16))
+        self.frame.set("Payload", payload)
+
+        with open("tests/ethernet_ipv4_udp.raw", "rb") as file:
+            msg_as_bytes: bytes = file.read()
+
+        self.assertEqual(self.frame.to_bytes, msg_as_bytes)
+
+    def test_generating_ieee_802_3(self) -> None:
+
+        payload = (
+            b"\x45\x00\x00\x14\x00\x01\x00\x00\x40\x00\x7c\xe7"
+            b"\x7f\x00\x00\x01\x7f\x00\x00\x01\x00\x00\x00\x00"
+            b"\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00"
+            b"\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00"
+        )
+
+        self.frame.set("Destination", int("FFFFFFFFFFFF", 16))
+        self.frame.set("Source", int("0", 16))
+        self.frame.set("Type_Length_TPID", 46)
+        self.frame.set("Type_Length", 46)
+        self.frame.set("Payload", payload)
+
+        self.assertTrue(self.frame.valid_message)
+
+        with open("tests/ethernet_802.3.raw", "rb") as file:
+            msg_as_bytes: bytes = file.read()
+
+        self.assertEqual(self.frame.to_bytes, msg_as_bytes)
+
+    def test_generating_ethernet_2_vlan(self) -> None:
+
+        payload = (
+            b"\x45\x00\x00\x14\x00\x01\x00\x00\x40\x00\x7c\xe7"
+            b"\x7f\x00\x00\x01\x7f\x00\x00\x01\x00\x00\x00\x00"
+            b"\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00"
+            b"\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x0a"
+        )
+
+        self.frame.set("Destination", int("FFFFFFFFFFFF", 16))
+        self.frame.set("Source", int("0", 16))
+        self.frame.set("Type_Length_TPID", int("8100", 16))
+        self.frame.set("TPID", int("8100", 16))
+        self.frame.set("TCI", 1)
+        self.frame.set("Type_Length", int("0800", 16))
+        self.frame.set("Payload", payload)
+
+        self.assertTrue(self.frame.valid_message)
+
+        with open("tests/ethernet_vlan_tag.raw", "rb") as file:
+            msg_as_bytes: bytes = file.read()
+
+        print(self.frame.to_bytes.hex())
+        print(msg_as_bytes.hex())
+
+        self.assertEqual(self.frame.to_bytes, msg_as_bytes)
+
+    def test_generating_ethernet_2_vlan_dynamic(self) -> None:
+        pass
+
+    # rflx-in_ethernet-test
+
+    def test_parsing_ipv4_in_ethernet(self) -> None:
+        pass
+
+    def test_generating_ipv4_in_ethernet(self) -> None:
+        pass
+
+    # reflx-in_ipv4_test
+
+    def test_parsing_udp_in_ipv4(self) -> None:
+        pass
+
+    def test_parsing_udp_in_ipv4_in_ethernet(self) -> None:
+        pass
```

Comment:
Empty tests should fail. Otherwise they might be forgotten later.

Reply:
Fixed in https://github.com/Componolit/RecordFlux/pull/198/commits/1ab7b6311e8bc2f82f25c3c4b96d70a135337223.